### PR TITLE
Bug 1862738 - Put credential management strings before translation

### PR DESCRIPTION
**Pull request by sarah541**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1862738

### CODE REVIEW

**Review by rocketsroger — APPROVED** [review]

File: fenix/app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1644,27 +1652,46 @@
     <string name="preferences_android_autofill_description">Fill usernames and passwords in other apps on your device.</string>
     <!-- Preference option for adding a login -->
     <string name="preferences_logins_add_login">Add login</string>
+    <!-- Preference option for adding a login -->
```
Comment (Delphine) [inline fenix/app/src/main/res/values/strings.xml:1655]: Nit: comment should say "for adding a password"

File: fenix/app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -1644,27 +1652,46 @@
...
+    <string name="preferences_logins_add_login_2" tools:ignore="UnusedResources">Add password</string>
 
     <!-- Preference for syncing saved logins in Fenix -->
     <string name="preferences_passwords_sync_logins">Sync logins</string>
+    <!-- Preference for syncing saved logins in Fenix -->
```
Comment (Delphine) [inline fenix/app/src/main/res/values/strings.xml:1660]: Nit: comment should say passwords and not logins here

File: fenix/app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -1644,27 +1652,46 @@
...
+    <string name="preferences_passwords_sync_logins_2" tools:ignore="UnusedResources">Sync passwords</string>
     <!-- Preference for syncing saved logins in Fenix, when not signed in-->
     <string name="preferences_passwords_sync_logins_across_devices">Sync logins across devices</string>
+    <!-- Preference for syncing saved logins in Fenix, when not signed in-->
```
Comment (Delphine) [inline fenix/app/src/main/res/values/strings.xml:1664]: Nit: should say passwords vs logins

File: fenix/app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -1644,27 +1652,46 @@
...
+    <string name="preferences_passwords_sync_logins_across_devices_2" tools:ignore="UnusedResources">Sync passwords across devices</string>
     <!-- Preference to access list of saved logins -->
     <string name="preferences_passwords_saved_logins">Saved logins</string>
+    <!-- Preference to access list of saved logins -->
```
Comment (Delphine) [inline fenix/app/src/main/res/values/strings.xml:1668]: Nit: again comment should say passwords here

File: fenix/app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -1644,27 +1652,46 @@
...
+    <string name="preferences_passwords_saved_logins_2" tools:ignore="UnusedResources">Saved passwords</string>
     <!-- Description of empty list of saved passwords. Placeholder is replaced with app name.  -->
     <string name="preferences_passwords_saved_logins_description_empty_text">The logins you save or sync to %s will show up here.</string>
+    <!-- Description of empty list of saved passwords. Placeholder is replaced with app name.  -->
+    <string name="preferences_passwords_saved_logins_description_empty_text_2" tools:ignore="UnusedResources">The passwords you save or sync to %s will be listed here. All passwords you save are encrypted.
+</string>
     <!-- Preference to access list of saved logins -->
     <string name="preferences_passwords_saved_logins_description_empty_learn_more_link">Learn more about Sync.</string>
+    <!-- Preference to access list of saved logins -->
```
Comment (Delphine) [inline fenix/app/src/main/res/values/strings.xml:1677]: Comment does not address string below (it's about sync, not accessing list of saved logins)

File: fenix/app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -1644,27 +1652,46 @@
...
+    <string name="preferences_passwords_saved_logins_description_empty_learn_more_link_2" tools:ignore="UnusedResources">Learn more about sync</string>
     <!-- Preference to access list of login exceptions that we never save logins for -->
     <string name="preferences_passwords_exceptions">Exceptions</string>
     <!-- Empty description of list of login exceptions that we never save logins for -->
     <string name="preferences_passwords_exceptions_description_empty">Logins and passwords that are not saved will be shown here.</string>
+    <!-- Empty description of list of login exceptions that we never save logins for. Parameter will be replaced by app name. -->
```
Comment (Delphine) [inline fenix/app/src/main/res/values/strings.xml:1683]: Nit: comment says "logins" but string is about passwords

File: fenix/app/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -1644,27 +1652,46 @@
...
+    <string name="preferences_passwords_exceptions_description_empty_2" tools:ignore="UnusedResources">%s won\'t save passwords for sites listed here.</string>
```
Comment (Delphine) [inline fenix/app/src/main/res/values/strings.xml:1684]: Again, you can just drop `\ `and directly use typographic quote here

File: fenix/app/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -1644,27 +1652,46 @@
...
     <!-- Description of list of login exceptions that we never save logins for -->
     <string name="preferences_passwords_exceptions_description">Logins and passwords will not be saved for these sites.</string>
+    <!-- Description of list of login exceptions that we never save logins for. Parameter will be replaced by app name. -->
+    <string name="preferences_passwords_exceptions_description_2" tools:ignore="UnusedResources">%s won\'t save passwords for these sites.</string>
```
Comment (Delphine) [inline fenix/app/src/main/res/values/strings.xml:1688]: same about apostrophe

File: fenix/app/src/main/res/values/strings.xml [hunk 9]
```diff
@@ -1644,27 +1652,46 @@
...
+    <string name="preferences_passwords_saved_logins_description_empty_learn_more_link_2" tools:ignore="UnusedResources">Learn more about sync</string>
     <!-- Preference to access list of login exceptions that we never save logins for -->
     <string name="preferences_passwords_exceptions">Exceptions</string>
     <!-- Empty description of list of login exceptions that we never save logins for -->
     <string name="preferences_passwords_exceptions_description_empty">Logins and passwords that are not saved will be shown here.</string>
+    <!-- Empty description of list of login exceptions that we never save logins for. Parameter will be replaced by app name. -->
+    <string name="preferences_passwords_exceptions_description_empty_2" tools:ignore="UnusedResources">%s won\'t save passwords for sites listed here.</string>
     <!-- Description of list of login exceptions that we never save logins for -->
     <string name="preferences_passwords_exceptions_description">Logins and passwords will not be saved for these sites.</string>
+    <!-- Description of list of login exceptions that we never save logins for. Parameter will be replaced by app name. -->
```
Comment (Delphine) [inline fenix/app/src/main/res/values/strings.xml:1687]: Nit: I think this should be about passwords too here, vs logins

File: fenix/app/src/main/res/values/strings.xml [hunk 10]
```diff
@@ -1644,27 +1652,46 @@
...
+    <string name="preferences_passwords_exceptions_description_2" tools:ignore="UnusedResources">%s won\'t save passwords for these sites.</string>
     <!-- Text on button to remove all saved login exceptions -->
     <string name="preferences_passwords_exceptions_remove_all">Delete all exceptions</string>
     <!-- Hint for search box in logins list -->
     <string name="preferences_passwords_saved_logins_search">Search logins</string>
+    <!-- Hint for search box in logins list -->
```
Comment (Delphine) [inline fenix/app/src/main/res/values/strings.xml:1693]: Nit: should be "passwords" and not logins

File: fenix/app/src/main/res/values/strings.xml [hunk 11]
```diff
@@ -1693,10 +1720,16 @@
     <string name="saved_login_hide_password">Hide password</string>
     <!-- Message displayed in biometric prompt displayed for authentication before allowing users to view their logins -->
     <string name="logins_biometric_prompt_message">Unlock to view your saved logins</string>
+    <!-- Message displayed in biometric prompt displayed for authentication before allowing users to view their logins -->
```
Comment (Delphine) [inline fenix/app/src/main/res/values/strings.xml:1723]: again should be passwords

File: fenix/app/src/main/res/values/strings.xml [hunk 12]
```diff
@@ -1713,6 +1746,8 @@
     <string name="saved_logins_sort_strategy_last_used">Last used</string>
     <!-- Content description (not visible, for screen readers etc.): Sort saved logins dropdown menu chevron icon -->
     <string name="saved_logins_menu_dropdown_chevron_icon_content_description">Sort logins menu</string>
+    <!-- Content description (not visible, for screen readers etc.): Sort saved logins dropdown menu chevron icon -->
```
Comment (Delphine) [inline fenix/app/src/main/res/values/strings.xml:1749]: This comment reads as a developer comment: "Sort saved logins dropdown menu chevron icon"
Localizers may not be familiar with technical jargon, so it's helpful to provide clear and straightforward comments that make it easy for them to understand the purpose and function of the string. I would drop that part out of the comment and just leave the first part

File: fenix/app/src/main/res/values/strings.xml [hunk 13]
```diff
@@ -1922,30 +1985,52 @@
...
     <string name="login_deletion_confirmation">Are you sure you want to delete this login?</string>
+    <!-- Message in delete confirmation dialog for logins -->
+    <string name="login_deletion_confirmation_2" tools:ignore="UnusedResources">Are you sure you want to delete this password?</string>
     <!-- Positive action of a dialog asking to delete  -->
     <string name="dialog_delete_positive">Delete</string>
     <!-- Negative action of a dialog asking to delete login -->
     <string name="dialog_delete_negative">Cancel</string>
     <!--  The saved login options menu description. -->
     <string name="login_options_menu">Login options</string>
+    <!--  The saved login options menu description. -->
```
Comment (Delphine) [inline fenix/app/src/main/res/values/strings.xml:1996]: Nit: should be 'password' in comment

File: fenix/app/src/main/res/values/strings.xml [hunk 14]
```diff
@@ -1922,30 +1985,52 @@
...
+    <string name="login_options_menu_2" tools:ignore="UnusedResources">Password options</string>
     <!--  The editable text field for a login's web address. -->
     <string name="saved_login_hostname_description">The editable text field for the web address of the login.</string>
+    <!--  The editable text field for a login's web address. -->
```
Comment (Delphine) [inline fenix/app/src/main/res/values/strings.xml:2000]: comment also needs update

File: fenix/app/src/main/res/values/strings.xml [hunk 15]
```diff
@@ -1922,30 +1985,52 @@
...
+    <string name="saved_login_hostname_description_2" tools:ignore="UnusedResources">The editable text field for the website address of the password.</string>
     <!--  The editable text field for a login's username. -->
     <string name="saved_login_username_description">The editable text field for the username of the login.</string>
+    <!--  The editable text field for a login's username. -->
```
Comment (Delphine) [inline fenix/app/src/main/res/values/strings.xml:2004]: again needs update

File: fenix/app/src/main/res/values/strings.xml [hunk 16]
```diff
@@ -1922,30 +1985,52 @@
...
     <string name="saved_login_password_description">The editable text field for the password of the login.</string>
+    <!--  The editable text field for a login's password. -->
+    <string name="saved_login_password_description_2" tools:ignore="UnusedResources">The editable text field for the password.</string>
     <!--  The button description to save changes to an edited login. -->
     <string name="save_changes_to_login">Save changes to login.</string>
+    <!--  The button description to save changes to an edited login. -->
+    <string name="save_changes_to_login_2" tools:ignore="UnusedResources">Save changes.</string>
     <!--  The page title for editing a saved login. -->
     <string name="edit">Edit</string>
+    <!--  The page title for editing a saved login. -->
```
Comment (Delphine) [inline fenix/app/src/main/res/values/strings.xml:2016]: comment needs update

File: fenix/app/src/main/res/values/strings.xml [hunk 17]
```diff
@@ -1922,30 +1985,52 @@
...
+    <string name="edit_2" tools:ignore="UnusedResources">Edit password</string>
     <!--  The page title for adding new login. -->
     <string name="add_login">Add new login</string>
+    <!--  The page title for adding new login. -->
```
Comment (Delphine) [inline fenix/app/src/main/res/values/strings.xml:2020]: comment needs update

**Review by Delphine — CHANGES_REQUESTED** [review]
Mostly seems like comment updates are needed (seems like many were copied directly from the previous existing string). When used, localization comments should describe the purpose and function of the immediate string they are attached to.
More about localization comments [here](https://mozilla-l10n.github.io/documentation/localization/dev_best_practices.html#add-localization-notes).

File: fenix/app/src/main/res/values/strings.xml [hunk 18]
```diff
@@ -1644,27 +1652,46 @@
...
+    <string name="preferences_passwords_saved_logins_2" tools:ignore="UnusedResources">Saved passwords</string>
     <!-- Description of empty list of saved passwords. Placeholder is replaced with app name.  -->
     <string name="preferences_passwords_saved_logins_description_empty_text">The logins you save or sync to %s will show up here.</string>
+    <!-- Description of empty list of saved passwords. Placeholder is replaced with app name.  -->
+    <string name="preferences_passwords_saved_logins_description_empty_text_2" tools:ignore="UnusedResources">The passwords you save or sync to %s will be listed here. All passwords you save are encrypted.
+</string>
     <!-- Preference to access list of saved logins -->
     <string name="preferences_passwords_saved_logins_description_empty_learn_more_link">Learn more about Sync.</string>
+    <!-- Clickable text for opening an external link for more information about Sync. -->
+    <string name="preferences_passwords_saved_logins_description_empty_learn_more_link_2" tools:ignore="UnusedResources">Learn more about sync</string>
```
Comment (Delphine) [inline fenix/app/src/main/res/values/strings.xml:1678]: I missed this! "Sync" should have a capital "S" (see for example string above)

**Review by Delphine — CHANGES_REQUESTED** [review]
Updates looking good thanks! just spotted one more update during this review

**Review by Delphine — APPROVED** [review]
Thanks for the fixes!

**Pull request merged**